eth/watchers/roundswatcher_test.go

```diff
@@ -0,0 +1,135 @@
+package watchers
+
+import (
+	"math/big"
+	"testing"
+	"time"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/livepeer/go-livepeer/eth"
+	"github.com/livepeer/go-livepeer/eth/blockwatch"
+	"github.com/livepeer/go-livepeer/pm"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSetAndGetLastInitializedRound(t *testing.T) {
+	rw := &RoundsWatcher{}
+	round := big.NewInt(5)
+	rw.setLastInitializedRound(round)
+	assert.Equal(t, rw.lastInitializedRound, round)
+	// there is no error path for LastInitializedRound so we don't have to test, it is always nil
+	// the return argument is there to adhere to the RoundsManager interface
+	r, err := rw.LastInitializedRound()
+	assert.Nil(t, err)
+	assert.Equal(t, r, round)
+}
+
+func TestSetAndGetBlockHashForRound(t *testing.T) {
+	assert := assert.New(t)
+	rw, err := NewRoundsWatcher(nil, &eth.StubClient{}, stubRoundsManagerAddr)
+	assert.Nil(err)
+	var round [32]byte
+	copy(round[:], "hello world")
+	rw.setBlockHashForRound(round)
+	assert.Equal(rw.blockHashForRound, round)
+	// round is nil returns an error
+	r, err := rw.BlockHashForRound(nil)
+	assert.Equal(r, [32]byte{})
+	assert.EqualError(err, "round is nil")
+
+	// round argument is not lastRound, make RPC call
+	rw.setLastInitializedRound(big.NewInt(5))
+	r, err = rw.BlockHashForRound(big.NewInt(0))
+	assert.Nil(err)
+	assert.Equal(r, [32]byte{})
+
+	// round argument is equal to lastInitializedRound, return blockhash in cache
+	r, err = rw.BlockHashForRound(big.NewInt(5))
+	assert.Nil(err)
+	assert.Equal(r, round)
+}
+
+func TestWatchAndStop(t *testing.T) {
+	assert := assert.New(t)
+	lpEth := &eth.StubClient{}
+	watcher := &stubBlockWatcher{}
+	rw, err := NewRoundsWatcher(watcher, lpEth, stubRoundsManagerAddr)
+	assert.Nil(err)
+
+	header := defaultMiniHeader()
+	round := [32]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 8}
```

Comment:
For consistency can we use a `newStubNewRoundLog()` method to return the log here? See [examples](https://github.com/livepeer/go-livepeer/pull/1048/files#diff-c5c00511e3b96fae6be6a993099cb207R43) of other stub logs that are pulled from mainnet

Reply:
fixed in 80d3e25